core/domain/classifier_services.py

```diff
@@ -25,38 +25,51 @@
 from core import feconf
 from core.domain import classifier_domain
 from core.domain import config_domain
+from core.domain import exp_domain
 from core.domain import exp_fetchers
 from core.domain import fs_services
 from core.platform import models
 
-(classifier_models, exp_models) = models.Registry.import_models(
-    [models.NAMES.classifier, models.NAMES.exploration])
+from typing import Dict, List, Optional, Sequence, Union
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import classifier_models
+
+(classifier_models,) = models.Registry.import_models([models.NAMES.classifier])
 
 
 # NOTE TO DEVELOPERS: This function should be kept in sync with its counterpart
 # in Oppia-ml.
-def generate_signature(secret, message, vm_id):
+def generate_signature(
+    secret: bytes,
+    message: Union[str, bytes],
+    vm_id: str
+) -> str:
     """Generates digital signature for given data.
 
     Args:
         secret: bytes. The secret used to communicate with Oppia-ml.
-        message: bytes. The message payload data.
+        message: bytes|str. The message payload data.
         vm_id: str. The ID of the VM that generated the message.
 
     Returns:
         str. The signature of the payload data.
     """
-    if isinstance(vm_id, str):
-        vm_id = vm_id.encode('utf-8')
+    converted_vm_id: Union[str, bytes] = vm_id
+    if isinstance(converted_vm_id, str):
+        converted_vm_id = converted_vm_id.encode('utf-8')
```

Comment:
If we do not introduce a new variable here with the type `Union[str, bytes]` and just use `vm_id` as `str` type in `generate_signature` method, then MyPy throws an error at line:
```python
    if isinstance(vm_id, str):
        vm_id = vm_id.encode('utf-8')
```
because here `vm_id` is assigned with `bytes` type value and previously its type was `str`.

So, to tackle this I introduced a new variable here but the other possible approach is annotating `vm_id` as `Union[str, bytes]`. I stick to `str` type for `vm_id` because in the whole codebase we are never providing `bytes` value to this function's `vm_id` argument.

Reply:
If clause is removed and assert is added !